servicetalk-buffer-api/src/main/java/io/servicetalk/buffer/api/CharSequences.java

```diff
@@ -206,39 +206,81 @@ public static int caseInsensitiveHashCode(CharSequence seq) {
     }
 
     /**
-     * Split a given {@link CharSequence} to separate ones on the given {@code delimiter}.
-     * The returned {@link CharSequence}s are created by invoking the {@link CharSequence#subSequence(int, int)} method
-     * on the main one.
+     * Split a given {@link #newAsciiString(Buffer) AsciiString} to separate ones on the given {@code delimiter}.
      *
+     * Trimming white-space before and after each token can be controlled by the {@code trim} flag
      * This method has no support for regex.
      *
-     * @param input The initial {@link CharSequence} to split, this experiences no side effects
-     * @param delimiter The delimiter character
+     * @param input The initial {@link CharSequence} to split, this experiences no side effects.
+     * @param delimiter The delimiter character.
+     * @param trim Flag to control whether the individual items must be trimmed.
      * @return a {@link List} of {@link CharSequence} subsequences of the input with the separated values
      */
-    public static List<CharSequence> split(final CharSequence input, final char delimiter) {
+    public static List<CharSequence> split(final CharSequence input, final char delimiter, final boolean trim) {
         if (input.length() == 0) {
             return emptyList();
         }
 
+        boolean isAscii = isAsciiString(input);
+        List<CharSequence> result = new ArrayList<>(4);
+
         int startIndex = 0;
-        List<CharSequence> result = new ArrayList<>();
-        for (int i = 0; i < input.length(); i++) {
-            if (input.charAt(i) == delimiter) {
-                if ((i - startIndex) > 0) {
-                    result.add(input.subSequence(startIndex, i));
+        int endIndex = -1;
+
+        if (trim) {
+            startIndex = -1;
```

Comment:
consider putting each variant in its own method. which has the following benefits:
- reduces scope so that only the necessary state is defined
- easier inlining for JVM -> reduces the overall byte code of the containing method and makes it easier for the JVM to inline this method if the `trim` value is mostly the same.